### ISSUE TITLE
Preprocessing on embedding training dataset

### ISSUE THREAD

**Issue by zisding** [code2vec-critique/generalizability#3]
Hi, 

Thanks for opening source this, it's great work.

However, I encountered some problems when I wanted to reproduce the results. I went through these 3 tasks folders and didn't find the preprocess scripts. Actually, this step is pretty important for training the embeddings/reproduce the results in this paper, I suppose.

At your convenience, If you can please share the preprocessing scripts used in the paper it would be really helpful.

Thanks in advance

**Comment by kanghj** [comment]
Hi, thanks for your interest in this work!

Do you mean the preprocess scripts for the code authorship task? Let me try to update the instructions on the readme to be more useful. Sorry for any inconvenience caused by my late response.

**Comment by zisding** [comment]
Thank you for your reply. 

It would be much better if we can update our readme of the tasks. 😁

Besides, I am a little curious about the scripts for preprocessing the original java dataset to the corpus for Glove. Do we remove the comments/keywords/punctuations before we train the embeddings?

Thank you.

**Comment by kanghj** [comment]
> Besides, I am a little curious about the scripts for preprocessing the original java dataset to the corpus for Glove. Do we remove the comments/keywords/punctuations before we train the embeddings?

We did not remove comments/keywords/punctuations before training the glove embeddings, but I ran some tokenizer on it. I tuned the hyper-parameters to have a similar vocabulary size as the code2vec embeddings. I did some ad-hoc, small-scale experiments on the window size and other parameters, but I found from my limited results that they didn't really influence the results on the task I experimented on. 

Just FYI, I became aware of a paper "Semantic Source Code Models
Using Identifier Embeddings" by Vasiliki et al. in MSR 2019. Their embeddings trained using fastText is publicly available. They preprocessed their embeddings carefully, perhaps this may be useful for you.
